docs/contributing/database-migrations/mssql.md

```diff
@@ -0,0 +1,109 @@
+---
+sidebar_position: 1
+---
+
+# MSSQL
+
+:::info
+
+For instructions on how to apply database migrations, please refer to the
+[Getting Started](../../getting-started/server/database/mssql/index.md) documentation.
+
+:::
+
+## SQL Database project
+
+:::warning
+
+TODO: Update
+
+:::
+
+We use a [SQL Database project][SSDT] (`sqlproj`) to develop the database locally. This means we
+have an up-to-date representation of the database in `src/Sql`, and any modifications needs to be
+represented there as well. These projects behave slightly different depending on which OS you are
+using.
+
+=== "Windows" Visual Studio provides built in support for Database projects with their SQL Server
+Data Tools. This is usually the optimal development environment with built in support for schema
+comparison and much more.
+
+=== "Mac & Linux" Visual Studio for Mac unfortunately does not support Database projects, however
+you may instead use [Visual Studio Code](vscode) or [Azure Data Studio](azureds) with the
+[SQL Database Projects](SDPE) extension, which provides schema comparison and more. You may also
+modify the `.sql` files directly with any text editor.
+
+    Do note that when adding or renaming SQL files you might need to manually update the references
+    in the `.sqlproj` file.
+
+To make a database change, start by modifying the `.sql` files in `src/Sql/dbo`. These changes will
+also need to be applied in a migration script. Migration scripts are located in
+`util/Migrator/DbScripts`.
+
+You can either generate the migration scripts automatically using the _Schema Comparison_
+functionality or by manually writing them. Do note that the automatic method will only take you so
+far and it will need to be manually edited to adhere to the code styles.
+
+## Modifying the database
+
+Since we follow [Evolutionary Database Design _(EDD)_](./edd.mdx), any migration that modifies
+existing columns most likely needs to be split into at least two parts: a backwards compatible
+transition phase, and a non-backwards compatible phase.
+
+### Best Practices
+
+When writing a migration script there are a couple of best practices we follow. Please check the
+[T-SQL Code Style][code-style-sql] for more details. But the most important aspect is ensuring the
+script can be re-run on the database multiple times without producing any errors or data loss.
+
+### Backwards Compatible
```

Comment:
We should add a section better describing when non backwards compatible migrations needs to be used.